contracts/HATVault.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+// Disclaimer https://github.com/hats-finance/hats-contracts/blob/main/DISCLAIMER.md
+
+pragma solidity 0.8.14;
+
+
+import "./vaults/Claim.sol";
+import "./vaults/Deposits.sol";
+import "./vaults/Params.sol";
+import "./vaults/Withdrawals.sol";
+
+contract HATVault is Claim, Deposits, Params, Withdrawals {
+
+    function _deposit(
+        address caller,
+        address receiver,
+        uint256 assets,
+        uint256 shares
+    ) internal override(Deposits, ERC4626Upgradeable) {
+        Deposits._deposit(caller, receiver, assets, shares);
+    }
+
+    function _beforeTokenTransfer(
+        address from,
+        address to,
+        uint256 amount
+    ) internal virtual override {
+        if (from == address(0) || to == address(0)) {
+            return;
+        }
+        // Users can only deposit for themselves if withdraw request exists
+        if (withdrawEnableStartTime[to] != 0) {
+            revert CannotDepositToAnotherUserWithWithdrawRequest();
```

Comment:
not the best error here, should be `CannotTransferToAnotherUserWithWithdrawRequest`

Reply:
Also this does not cover the case that the withdrawrequest has timed out.

Reply:
Good point